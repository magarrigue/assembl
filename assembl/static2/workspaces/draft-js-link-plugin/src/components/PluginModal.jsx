// @flow
import * as React from 'react';

type Props = {
  children: React.Node,
  close: (SyntheticEvent<HTMLButtonElement>) => void,
  closeLabel: string,
  title: string
};

const PluginModal = ({ children, close, closeLabel, title }: Props) => (
  <div>
    <div className="modal-backdrop fade in" />
    <div className="insertion-box box">
      <div className="modal-header">
        <button title={closeLabel} onClick={close} className="close">
          <span className="assembl-icon-cancel" />
        </button>
        <h4 className="modal-title">{title}</h4>
      </div>
      <div className="modal-body">{children}</div>
    </div>
  </div>
);

export default PluginModal;