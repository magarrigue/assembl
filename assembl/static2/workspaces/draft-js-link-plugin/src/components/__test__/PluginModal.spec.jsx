import React from 'react';
import renderer from 'react-test-renderer';

import PluginModal from '../PluginModal';

describe('PluginModal component', () => {
  it('should render a modal', () => {
    const closeSpy = jest.fn();
    const props = {
      children: <div>Try to reboot the TCP pixel!</div>,
      close: closeSpy,
      closeLabel: 'Close',
      title: 'My modal'
    };
    const component = renderer.create(<PluginModal {...props} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});