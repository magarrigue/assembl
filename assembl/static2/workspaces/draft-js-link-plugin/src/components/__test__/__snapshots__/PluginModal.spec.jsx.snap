// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PluginModal component should render a modal 1`] = `
<div>
  <div
    className="modal-backdrop fade in"
  />
  <div
    className="insertion-box box"
  >
    <div
      className="modal-header"
    >
      <button
        className="close"
        onClick={[Function]}
        title="Close"
      >
        <span
          className="assembl-icon-cancel"
        />
      </button>
      <h4
        className="modal-title"
      >
        My modal
      </h4>
    </div>
    <div
      className="modal-body"
    >
      <div>
        Try to reboot the TCP pixel!
      </div>
    </div>
  </div>
</div>
`;
